### DISCUSSION TITLE
Rebinding SCM Widget

### DISCUSSION THREAD

**Discussion by maheshrajrp** [eclipse-theia/theia#12339] [Q&A]
Hi,
I'm currently working on enhancements related to SCM Widget, I made few changes in SCM Widget and tried to rebind. 
I've tried rebinding my custom widget like this and am facing an issue. 
```
    bind(CustomScmTreeWidget).toSelf().inSingletonScope();
    rebind(ScmTreeWidget).to(CustomScmTreeWidget).inSingletonScope();
```
And am getting this issue. 
![image](https://user-images.githubusercontent.com/80001447/227005534-4c8975b3-0696-46d2-88db-ba371a2da90d.png)

It would be of great help if someone could help regarding this issue.

Thanks in advance.

**Comment by maheshrajrp** [comment]
@msujew Can you assist with this case ?

**Comment by msujew** [comment]
@maheshrajrp As you can see in the original binding:

https://github.com/eclipse-theia/theia/blob/441c5a1cf432f29449eab2919ebc5781d0ea8787/packages/scm/src/browser/scm-frontend-module.ts#L67-L70

The `ScmTreeWidget` is using a different binding mechanism compared to most services. I believe you should be able to get there by doing something like this:

```ts
bind(WidgetFactory).toDynamicValue(({ container }) => ({
  id: ScmTreeWidget.ID,
  createWidget: () => container.get(CustomScmTreeWidget)
})).inSingletonScope();
```

**Reply by maheshrajrp** [reply]
Hi @msujew ,
This starts up Theia Frontend, however, `CustomWidget` isn't being picked up, still old widget is being picked up. Should we unbind old widget ?
